<HTML>
<BODY>
<PRE>
<!-- Manpage converted by man2html 3.0.1 -->
miranda(l)             The miRanda Package             miranda(l)




</PRE>
<H2>NAME</H2><PRE>
       miranda  -  Finds  potential  target  sites  for miRNAs in
       genomic sequence


</PRE>
<H2>SYNOPSIS</H2><PRE>
       Basic Usage:
       <B>miranda</B> <I>file1</I> <I>file2</I> [ <I>options</I> ... ]


       Advanced:
       <B>miranda</B>    <I>file1</I>    <I>file2</I>     [-sc <I>score</I>]     [-en <I>energy</I>]
       [-scale <I>scale</I>]  [-loose]  [-go <I>X</I>]  [-ge <I>Y</I>]  [-out <I>fileout</I>]
       [-quiet]  [-trim <I>T</I>]  [-noenergy]  [-shuffle]   [-s <I>nshuff</I>]
       [-w <I>W</I>] [-uniform] [-z <I>zscore</I>]



</PRE>
<H2>DESCRIPTION</H2><PRE>
       <B>miRanda</B>  is  an  algorithm  for the detection of potential
       microRNA target sites in genomic sequences. <B>miRanda</B>  reads
       RNA  sequences  (such as microRNAs) from <B>file1</B> and genomic
       DNA/RNA sequences from <B>file2</B>. Both of these  files  should
       be  in FASTA format. This is an example of a FASTA format­
       ted sequence:

       &gt;embl|AJ550546|DME550546 Drosophila melanogaster  microRNA
       miR-bantam
       GTGAGAUCAUUUUGAAAGCUG

       One or more miRNA sequences from <B>file1</B> are scanned against
       all sequences in <B>file2</B>  and  potential  target  sites  are
       reported.  Potential  target  sites are identified using a
       two-step strategy.   First  a  dynamic  programming  local
       alignment  is carried out between the query miRNA sequence
       and  the  reference  sequence.  This  alignment  procedure
       scores  based  on  sequence  complementarity  and  not  on
       sequence identity.  In other words we look for A:U and G:C
       matches  instead of A:A, G:G, etc.  The G:U wobble bair is
       also permitted, but generally scores less  than  the  more
       optimal matches. Here is an example alignment:

           <B>Query:</B>  3' GTCGAAAGTTTTACTAGAGTG 5' (eg. miRNA)
                      <B>:||:|||||</B> <B>||||||||||</B>
           <B>Ref:</B>    5' TAGTTTTCACAATGATCTCGG 3' (eg. 3'UTR)

       The  second  phase  of  the  algorithm  takes high-scoring
       alignments (Those above a score threshold, defined by <B>-sc</B>)
       detected from phase 1 and estimates the thermodynamic sta­
       bility of RNA duplexes based on  these  alignments.   This
       second  phase of the method utilizes folding routines from
       the <B>RNAlib</B> library, which is part of the <B>ViennaRNA</B> package
       written  by Ivo Hofacker. At this stage we generate a con­
       strained fictional single-stranded  RNA  composed  of  the
       query  sequence,  a  linker  and  the  reference  sequence
       (reversed). This structure then folded  using  <B>RNAlib</B>  and
       Optionally  some  rudimentary <B>statistics</B> about each target
       site can be generated by performing a number of alignments
       using shuffled reference sequences (see <B>-shuffled</B> option).
       A distribution is built from these  data  and  statistical
       parameters from this distribution are used to produce a Z-
       Score for a detected target site.



</PRE>
<H2>OPTIONS</H2><PRE>
       <B>--help</B> <B>-h</B>
              Displays help, usage information  and  command-line
              options.

       <B>--version</B> <B>-v</B> <B>--license</B>
              Display version and license information.

       <B>-sc</B> <I>score</I>
              Set  the  score threshold to <B>score</B>. Only alignments
              with scores &gt;= <B>score</B> will be used for further anal­
              ysis.

       <B>-en</B> <I>energy</I>
              Set the energy threshold to <B>energy</B>. Only alignments
              with energies &lt;= <B>energy</B> will be  used  for  further
              analysis. This value should be negative.

       <B>-scale</B> <I>scale</I>
              Set the scaling parameter to <B>scale</B>. This scaling is
              applied to match / mismatch scores in the  critical
              10bp  region  of  the  5' end of the microRNA. Many
              known examples of miRNA:Target duplexes are  highly
              complementary in this region. This parameter can be
              thought of as a contrast function  to  more  effec­
              tively detect alignments of this type.

       <B>-loose</B> Remove  strict alignment heuristics. In normal mode
              heuristics are applied to predicted  duplexes  that
              count  the  number  of  complementary base-pairs in
              different regions of  the  alignment  according  to
              observations of known miRNA:target duplexes. Align­
              ments that fail this test are not displayed Turning
              this option off is less conservative but may result
              in more false-positive detections.

       <B>-go</B> <I>X</I>  Set the gap-opening penalty to  <B>X</B>  for  alignments.
              This value should be negative.

       <B>-ge</B> <I>Y</I>  Set  the  gap-extend  penalty  to <B>Y</B> for alignments.
              This value should be negative.

       <B>-out</B> <I>fileout</I>
              Print results to an output file called <B>fileout</B>.

       <B>-quiet</B> Quiet mode, produces the minimum of output.

       <B>-trim</B> <I>T</I>
              Trim reference sequences to <B>T</B>  nucleotides.  Useful
              when  using  noisy  predicted  3'UTRs  as reference

       <B>-shuffle</B>
              For  each  analysis between a microRNA and a refer­
              ence sequence, also do  <B>nshuff</B>  alignments  between
              the microRNA and shuffled reference sequences. This
              is used to build a distribution of shuffled  align­
              ment scores that can be used to estimate the relia­
              bility of a given score.  This analysis produces  a
              <B>Z-Score</B>  for  each predicted target. It is possible
              to filter results using  this  score  with  the  <B>-z</B>
              option.   Shuffling statistics are optional and are
              not turned on by default. It should also  be  noted
              that  this  dramatically  increase  the  number  of
              alignments processed, and will slow  the  algorithm
              down considerably.

       <B>-s</B> <I>nshuff</I>
              Set  the total number of random shuffle analyses to
              <B>nshuff</B>. Larger numbers of analyses produces a  more
              reliable <B>Z-score</B>.

       <B>-w</B> <I>W</I>   For  window-based  shuffling the window size can be
              changed by modifying the <B>W</B> parameter. Window  based
              shuffling  maintains  nucleotide composition across
              the sequence, shuffling takes place in a small win­
              dow that moves across the sequence.

       <B>-uniform</B>
              Disables windowed shuffling, and shuffles uniformly
              across the reference sequence.

       <B>-z</B> <I>zscore</I>
              Sets a <B>Z-Score</B> threshold for all targets  based  on
              sequence  shuffling  statistics. Only detected tar­
              gets with <B>Z-Scores</B> &gt;= <B>zscore</B> will be  displayed  by
              the algorithm.



</PRE>
<H2>REFERENCES</H2><PRE>
       If  you  use  this  program  for your research then please
       include the following citation:

       A.J. Enright, B. John, U. Gaul, T. Tuschl, C. Sander, D.S.
       Marks; (2003)
       MicroRNA targets in Drosophila; Genome Biology <B>5(1)</B>:R1.

       RNAlib Citations:

       I.L. Hofacker, W. Fontana, P.F. Stadler, S. Bonhoeffer, M.
       Tacker, P. Schuster (1994) Fast Folding and Comparison  of
       RNA  Secondary  Structures.   Monatshefte  f.  Chemie 125:
       167-188

       M. Zuker, P. Stiegler (1981) Optimal computer  folding  of
       large  RNA  sequences  using  thermodynamic  and auxiliary
       information, Nucl Acid Res 9: 133-148

       J.S. McCaskill (1990) The equilibrium  partition  function
       and  base  pair  binding  probabilities  for RNA secondary


</PRE>
<H2>BUGS</H2><PRE>
       Comments    and    bug-reports    should    be   sent   to
       miranda@cbio.mskcc.org.



Anton Enright                  1.0                     miranda(l)
</PRE>
<HR>
<ADDRESS>
Man(1) output converted with
<a href="http://www.oac.uci.edu/indiv/ehood/man2html.html">man2html</a>
</ADDRESS>
</BODY>
</HTML>
